lib/rules/no-constant-condition.js

```diff
@@ -115,15 +118,74 @@ module.exports = {
         }
 
         /**
-         * Reports when the given node contains a constant condition.
+         * Tracks when the given node contains a constant condition.
          * @param {ASTNode} node The AST node to check.
          * @returns {void}
          * @private
          */
         function checkConstantCondition(node) {
             if (node.test && isConstant(node.test, true)) {
+                setOfConstantLoops.add(node);
+            }
+        }
+
+        /**
+         * Reports when the set contains the given constant condition node
+         * @param {ASTNode} node The AST node to check.
+         * @returns {void}
+         * @private
+         */
+        function checkConstantConditionLoopInSet(node) {
+            if (setOfConstantLoops.has(node)) {
+                setOfConstantLoops.delete(node);
+                context.report({ node, message: "Unexpected constant condition." });
+            }
+        }
+
+        /**
+         * Reports when the given node contains a constant condition.
+         * @param {ASTNode} node The AST node to check.
+         * @returns {void}
+         * @private
+         */
+        function checkIfConstant(node) {
+            if (node.test && isConstant(node.test, true)) {
+                context.report({ node, message: "Unexpected constant condition." });
+            }
+        }
+
+        /**
+         * Clears setOfConstantLoops
+         * @returns {void}
+         * @private
+         */
+        function clearSetOfLoops() {
```

Comment:
I don't feel strongly about this, but personally it seems unnecessary to me to have a function for this -- in my opinion, it adds unnecessary indirection.